src/PSR7/Validators/PathValidator.php

```diff
@@ -39,14 +40,21 @@ public function validate(OperationAddress $addr, MessageInterface $message) : vo
     /**
      * @throws InvalidPath
      * @throws NoPath
+     * @throws NoParameter
      */
     private function validateRequest(OperationAddress $addr, RequestInterface $message) : void
     {
         $specs = $this->finder->findPathSpecs($addr);
 
         $path             = $message->getUri()->getPath();
         $pathParsedParams = $addr->parseParams($path); // ['id'=>12]
-        $validator        = new SchemaValidator($this->detectValidationStrategy($message));
+        if ($pathParsedParams===[]) {
+            return;
+        }
+        if ($specs === []) {
+            throw NoParameter::fromPath($addr->path());
+        }
+        $validator = new SchemaValidator($this->detectValidationStrategy($message));
 
         foreach ($pathParsedParams as $name => $value) {
```

Comment:
It looks like that this PR doesn't address the problem described in the original issue. If we have two path parameters, and one of them is described and a second one is not - we will still reach this point having `$specs[$name]` index problems.

Having no specs from non-empty params is bad for sure but the problem that needs to be fixed is that we are missing specific paramter in schema, an this is not checked.